src/AggregateFunctions/TimeSeries/AggregateFunctionTimeseriesChanges.h

```diff
@@ -0,0 +1,203 @@
+        ValueType & result, UInt8 & null) const
+    {
+        if (samples_in_window.empty())
+        {
+            result = 0;
+            null = 1;
+            return;
+        }
+
+        UInt64 change_count = 0;
+        std::pair<TimestampType, ValueType> prev_sample = samples_in_window.front();
+        for (const auto& sample : samples_in_window)
+        {
+            if ((is_resets && sample.second < prev_sample.second) || (!is_resets && sample.second != prev_sample.second))
+                change_count++;
+            prev_sample = sample;
+        }
+
+        result = static_cast<ValueType>(change_count);
+        null = 0;
+    }
+
+public:
+    /// Insert the result into the column
+    void doInsertResultInto(AggregateDataPtr __restrict place, IColumn & to) const
+    {
+        ColumnArray & arr_to = typeid_cast<ColumnArray &>(to);
+        ColumnArray::Offsets & offsets_to = arr_to.getOffsets();
+
+        offsets_to.push_back(offsets_to.back() + Base::bucket_count);
+
+        if (!Base::bucket_count)
+            return;
+
+        ColumnNullable & result_to = typeid_cast<ColumnNullable &>(arr_to.getData());
+        auto & data_to = typeid_cast<typename Base::ColVecResultType &>(result_to.getNestedColumn()).getData();
+        auto & nulls_to = result_to.getNullMapData();
+
+        const size_t old_size = data_to.size();
+        chassert(old_size == nulls_to.size(), "Sizes of nested column and null map of Nullable column are not equal");
+
+        data_to.resize(old_size + Base::bucket_count);
+        nulls_to.resize(old_size + Base::bucket_count);
+
+        ValueType * values = data_to.data() + old_size;
+        UInt8 * nulls = nulls_to.data() + old_size;
+
+        const auto & buckets = Base::data(place)->buckets;
+
+        std::deque<std::pair<TimestampType, ValueType>> samples_in_window;
+        std::vector<std::pair<TimestampType, ValueType>> timestamps_buffer;
+
+
+        /// Fill the data for missing buckets
+        for (UInt32 i = 0; i < Base::bucket_count; ++i)
+        {
+            const TimestampType current_timestamp = Base::start_timestamp + i * Base::step;
+
+            auto bucket_it = buckets.find(i);
+            if (bucket_it != buckets.end())
+            {
+                timestamps_buffer.clear();
+                /// Sort samples from the current bucket
+                for (const auto & [timestamp, value] : bucket_it->second.samples)
+                    timestamps_buffer.emplace_back(timestamp, value);
+                std::sort(timestamps_buffer.begin(), timestamps_buffer.end());
+
+                /// Add samples from the current bucket
+                for (const auto & [timestamp, value] : timestamps_buffer)
+                    samples_in_window.push_back({timestamp, value});
+            }
+
+            /// Remove samples that are out of the window
+            while (!samples_in_window.empty() && samples_in_window.front().first + Base::window < current_timestamp)
```

Comment:
Please replace `Base::window < current_timestamp` with `Base::window <= current_timestamp` according to https://github.com/ClickHouse/ClickHouse/pull/86588.